src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -614,63 +651,24 @@ static void TryAddStringField(Dictionary<string, string> extendedAttributes, str
             }
         }
 
-        // Generates an extended attribute key value pair string saved into a byte array, following the ISO/IEC 10646-1:2000 standard UTF-8 encoding format.
-        // https://pubs.opengroup.org/onlinepubs/9699919799/utilities/pax.html
-        private static byte[] GenerateExtendedAttributeKeyValuePairAsByteArray(byte[] keyBytes, byte[] valueBytes)
-        {
-            // Assuming key="ab" and value="cdef"
-
-            // The " ab=cdef\n" attribute string has a length of 9 chars
-            int suffixByteCount = 3 + // leading space, equals sign and trailing newline
-                keyBytes.Length + valueBytes.Length;
-
-            // The count string "9" has a length of 1 char
-            string suffixByteCountString = suffixByteCount.ToString();
-            int firstTotalByteCount = Encoding.ASCII.GetByteCount(suffixByteCountString);
-
-            // If we prepend the count string length to the attribute string,
-            // the total length increases to 10, which has one more digit
-            // "9 abc=def\n"
-            int firstPrefixAndSuffixByteCount = firstTotalByteCount + suffixByteCount;
-
-            // The new count string "10" has an increased length of 2 chars
-            string prefixAndSuffixByteCountString = firstPrefixAndSuffixByteCount.ToString();
-            int realTotalCharCount = Encoding.ASCII.GetByteCount(prefixAndSuffixByteCountString);
-
-            byte[] finalTotalCharCountBytes = Encoding.ASCII.GetBytes(prefixAndSuffixByteCountString);
-
-            // The final string should contain the correct total length now
-            List<byte> bytesList = new();
-
-            bytesList.AddRange(finalTotalCharCountBytes);
-            bytesList.Add(TarHelpers.SpaceChar);
-            bytesList.AddRange(keyBytes);
-            bytesList.Add(TarHelpers.EqualsChar);
-            bytesList.AddRange(valueBytes);
-            bytesList.Add(TarHelpers.NewLineChar);
-
-            Debug.Assert(bytesList.Count == (realTotalCharCount + suffixByteCount));
-
-            return bytesList.ToArray();
-        }
-
         // The checksum accumulator first adds up the byte values of eight space chars, then the final number
         // is written on top of those spaces on the specified span as ascii.
         // At the end, it's saved in the header field and the final value returned.
-        internal int WriteChecksum(int checksum, Span<byte> buffer)
+        internal static int WriteChecksum(int checksum, Span<byte> buffer)
```

Comment:
@carlossanlop,  even before my changes,  is this method functionally correct? The input and output spans are the same length,  but the output span has two characters at the end reserved,  so are we frequently losing digits from the checksum of the checksum is large enough?

Reply:
I'd have to investigate this with a header whose fields have ascii characters with the highest possible values, so that when they get added up for the checksum, its final value would go beyond the checksum field length, excluding the two reserved characters at the end.